### DISCUSSION TITLE
allowEIO3 - Socket.io v4 argument

### DISCUSSION THREAD

**Discussion by mmilosevic** [miguelgrinberg/Flask-SocketIO#1703] [Q&A] [answered]
I have a question regarding the `allowEIO3` Socket.io v4 configuration option. 

We have a client (an iOS app) that is using [Swift client library](https://github.com/socketio/socket.io-client-swift) v15 and the server with the following setup
* flask-socketio==4.3.2
* python-engineio==3.13.2
* python-socketio==4.6.1
 
With this configuration on the client and backend, everything works fine, but :).

We tried to upgrade `flask-socketio` to version 5.x sometime in January and we couldn't. `python-socketio` version used by `flask-socketio` 5.x and Swift client v15 could not communicate (EIO param on a handshake was omitted [issue](https://github.com/socketio/socket.io-client-swift/issues/1314)). 
Obviously, we weren't the only one ([The Socket.IO v4 server will now infer EIO=3 if the EIO query param is not included](https://socket.io/docs/v4/migrating-from-3-x-to-4-0/#configuration)).

I am trying to do the upgrade process now and I still cannot make it work. My setup is 
* client - still Swift client library v15
* backend - flask-socketio==5.1.1, python-engineio==4.2.1, python-socketio==5.4.1

Can someone tell me if the Swift client library (v15) and the latest `flask-socketio` version can work together if I set `allowEIO3` to True? And if so, to provide me an example of how to initialize the Flask-SocketIO server properly? Thanks!

**Comment by miguelgrinberg** [comment, marked as answer]
The python-socketio package does not have dual EIO 3 and 4 support, you have to pick the one that you need. The 5.x supports EIO=4 only. The 4.x supports EIO=3. This is in the version compatibility table in the documentation.

**Reply by mmilosevic** [reply]
Thanks. Is there a chance that we will get a dual support at some point?

**Reply by miguelgrinberg** [reply]
I don't really have it as a priority, so it is not in my task list for the near future. I think it would make more sense to ask the developer of the client to upgrade to EIO=4.

**Reply by mmilosevic** [reply]
We can update the client, that is not a problem, problem are all older mobile applications after we upgrade the backend.
